whois-api/src/main/java/net/ripe/db/whois/api/rest/RestServiceHelper.java

```diff
@@ -154,6 +156,9 @@ public static WebApplicationException createWebApplicationException(final Runtim
             }
             messages.addAll(queryException.getMessages());
 
+        } else if (exception instanceof IllegalArgumentException) {
+            responseBuilder = Response.status(Response.Status.BAD_REQUEST);
+            messages.add(QueryMessages.tooManyArguments());
```

Comment:
Can't there be other reasons why an IllegalArgumentException is thrown?

Reply:
Can you catch the exception closer to where it happens and add a global error there instead?

Reply:
updated